core/modules/qdisp/JobStatus.cc

```diff
@@ -0,0 +1,96 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2014 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include "qdisp/JobStatus.h"
+
+// System headers
+#include <iostream>
+
+namespace lsst {
+namespace qserv {
+namespace qdisp {
+// Static fields
+std::string const JobStatus::_empty;
+
+JobStatus::Info::Info(ResourceUnit const& resourceUnit_)
+    : resourceUnit(resourceUnit_),
+      state(UNKNOWN),
+      stateCode(0) {
+    stateTime = ::time(NULL);
+}
+
+
+std::ostream& operator<<(std::ostream& os, JobStatus::State const& state) {
+
+	std::map<JobStatus::State, std::string> state_message;
```

Comment:
maybe let's make this map static instead of redoing it for every operator<< call